### PR TITLE
Add: tests for oauth

### PR DESCRIPTION
**Pull request by Fraggle**
## Description

Add tests for oauth.
Most of the code was moved from bin/oauth.rs to src/oauth/app.rs so we could access it from the tests.

## Risk

None

## Deploy Plan

Deploy `oauth`

### CODE REVIEW

File: core/src/oauth/tests/test_credentials.rs [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    TestServer::new(app).unwrap()
+}
+
+#[derive(PartialEq)]
+enum HttpMethod {
+    GET,
+    POST,
+}
+
+async fn do_failing_api_call(url: String, method: HttpMethod, body: &serde_json::Value) -> String {
```
Comment (spolu) [inline core/src/oauth/tests/test_credentials.rs:25]: why repeat? ad not put in oauth/test/util.rs?

**Review by spolu — APPROVED** [review]
LGTM

I would maybe not repeat test_ in the name of the file and instead call them functional_xxx.rs ?

**Review by fontanierh — COMMENTED** [review]
Nice 🔥 

YPAHT but test now fail on CI  --  Would be really cool to run them on CI !

**Comment by Fraggle** [comment]
>YPAHT but test now fail on CI -- Would be really cool to run them on CI !

YPAHT but I'm on it :-)

**Comment by fontanierh** [comment]
> YPAHT but I'm on it :-)

Didn't have it ! nice

**Pull request merged**